serverlessworkflow/modules/ROOT/pages/persistence/postgresql-flyway-migration.adoc

```diff
@@ -0,0 +1,104 @@
+:flyway_baseline_migration_url: https://documentation.red-gate.com/fd/baseline-migrations-184127465.html
+
+Kogito provides an out-of-the-box way to manage your database schema changes with each upgrade. It uses link:{flyway_url}[Flyway] to achieve this functionality.
+
+When you upgrade your Kogito version, by default it will pick up the latest changes available from Kogito and apply them automatically.
+
+NOTE: This feature is currently available for only `kogito-addons-quarkus-persistence-postgresql` extension.
+
+== How to migrate
+
+=== Migrate using Flyway Config
+
+* Kogito provides a default mechanism for migrating your database while updating the Kogito version using the following `Flyway` properties (default value is `false`):
++
+--
+[source,console]
+----
+quarkus.flyway.migrate-at-start=true
+----
+--
+This will create a schema history table `flyway_schema_history` in your database to track the version of each database, recording in it every versioned migration file applied to build that version.
+
+=== Migrate using Flyway CLI
+* If you want to migrate manually you can use the Flyway migration CLI tool.
+
+** Download Flyway migration command-line tool from the link:{flyway_install_url}[Download Flyway Community Edition] link.
+** Add the following properties in the `/conf/flyway.conf` file:
++
+--
+[source,console]
+----
+flyway.url=jdbc:postgresql://localhost:5432/foobardb
+flyway.user=foo
+flyway.password=bar
+----
+--
+** You can specify these options with commands as well. The Flyway CLI will prompt for the username and password if they are missing in the configuration.
++
+--
+[source,console]
+----
+$ flyway migrate -url=jdbc:postgresql://localhost:5432/foobardb -user=foo -password=bar
+----
+--
+** Download the Kogito DDL zip from link:{kogito_ddl_script_url}[this link here] and extract it and use scripts from `postgresql` folder.
+** You can specify the location of the SQL files that need to be migrated using the `flyway.locations` option. For example,
++
+--
+[source,console]
+----
+flyway.locations=classpath:com.mycomp.migration,database/migrations,filesystem:/sql-migrations,s3:migrationsBucket,gcs:migrationsBucket
+----
+--
+
+=== Manually executing scripts
+You can use the provided SQL scripts in zip to migrate the database by executing them one by one.
+
+* You can find all Kogito DDL scripts by visiting link:{kogito_ddl_script_url}[this link here].
+* Download the zip file of the scripts from version same as your Kogito upgrade.
+* Extract the zip file.
+* Execute the `sql` files from `postgresql` folder.
+
+--
+[source,console]
+----
+> psql -H host -U username -d database_name -a -f create_table.sql
+----
+--
+
+== Baseline migration
+To automatically call baseline when migrate is executed against a non-empty schema with no metadata table, we can use the below property. This will baseline the schema with the baselineVersion before executing the migrations. Only migrations above baselineVersion will then be applied.
```

Comment:
_In order to apply migrations to an existing schema. It's necessary to establish the `baseline` from where Flyway should start executing new migrations. That means, in case your schema already contains a structure and from now on, new changes provided by Kogito should be applied via Flyway integration. In Flyway terms, only migrations above `baselineVersion` will then be applied. 
@akumar074 @radtriste does that sounds better?
I think it should also show the actual value users should give for that property. @akumar074 I believe that should be `1.32.0` ?

Reply:
@cristianonicolai Thanks. This looks much better. I updated it with `baseline-version`.